### PR TITLE
Rewrite the API

### PR DESCRIPTION
**Pull request by Borewit**
I don't think I can resolve merge conflicts on [stream-tokenizer](https://github.com/sindresorhus/file-type/tree/stream-tokenizer) via a PR, I can only create a PR with resolved conflicts to the master.

To resolve conflicts [stream-tokenizer](https://github.com/sindresorhus/file-type/tree/stream-tokenizer) the master branch needs to be merged directly onto [stream-tokenizer](https://github.com/sindresorhus/file-type/tree/stream-tokenizer).

### CODE REVIEW

**Comment by Borewit** [comment]
@sindresorhus can you please merge?
I got a few days left to support the transition if required.

**Comment by sindresorhus** [comment]
Done. Thanks for all your work on this 🙌

**Comment by sindresorhus** [comment]
Would you be interested in being added as a maintainer? No worries if not though.

File: index.d.ts [hunk 1]
```diff
@@ -1,309 +1,60 @@
...
+	@returns The detected file type and MIME type or `undefined` when there was no match.
+ */
+export function fromStream(stream: ReadableStream): Promise<core.FileTypeResult | undefined>;
 
-	/**
-	The minimum amount of bytes needed to detect a file type. Currently, it's 4100 bytes, but it can change, so don't hard-code it.
-	*/
-	readonly minimumBytes: number;
+/**
+	Deprecated: The minimum amount of bytes needed to detect a file type. Currently, it's 4100 bytes, but it can change, so don't hard-code it.
```
Comment (sindresorhus) [inline index.d.ts]: You say it's deprecated here, but it's not marked as such in the readme.

File: readme.md [hunk 2]
```diff
@@ -67,63 +90,147 @@ const fileType = require('file-type');
...
+const config = {
+	avoidHeadRequests: true
+};
+
+const audioTrackUrl = 'https://test-audio.netlify.com/Various%20Artists%20-%202009%20-%20netBloc%20Vol%2024_%20tiuqottigeloot%20%5BMP3-V2%5D/01%20-%20Diablo%20Swing%20Orchestra%20-%20Heroines.mp3';
 
-### fileType.stream(readableStream)
+(async () => {
+	const httpTokenizer = HttpTokenizer.fromUrl(audioTrackUrl, config);
+	await httpTokenizer.init();
```
Comment (sindresorhus) [inline readme.md]: This API is awkward. It should really just be:

```js
const httpTokenizer = await HttpTokenizer.fromUrl(audioTrackUrl, config);
```

**Comment by Borewit** [comment]
> Would you be interested in being added as a maintainer? 

Yes I am. Thank you for your cooperation, was a lot work on your end as well.

File: readme.md (hunk 2 above)
Comment (Borewit) [reply readme.md]: Just changed the API: #287

**Comment by Borewit** [comment]
Can we please release?

**Comment by sindresorhus** [comment]
> Yes I am.

Awesome! Added you to the repo now. While you have commit access to `master`, it's usually best to do PRs. It never hurts to have another set eye look at changes.

**Comment by sindresorhus** [comment]
> Can we please release?

Yes, will do.

**Comment by sindresorhus** [comment]
It's out: https://github.com/sindresorhus/file-type/releases/tag/v13.0.0

**Pull request merged**